iFixit/react-commerce
# Piwik: Update Ecommerce Events to V2

## Overview

We want to move away from piwik's ecommcer events and use their [ecommerce v2 events](https://developers.piwik.pro/en/latest/data_collection/web/javascript_tracking_client/api.html#e-commerce-v2), which are cleaner, easier, and more analogous to ga4 events. Also cleans up our analytics tracking by wrapping some of our analytics events in a joint function to call through to piwik and ga4. This in turns consolidates call spots for our analytics, making it simpler throughout the codebase.

Also makes sure the cart total price is accurate, as we weren't updating it previously.

## QA

This is a bit of a tedious one.
We changed some call sites for GA4 events, so we want to make sure view item, add to cart, and remove from cart are working still.

Then make sure that piwik events Product view, add to cart, remove from cart, cart update, and special events like frequently bought together and cross sells are all firing correctly.

Connects: https://github.com/iFixit/shopify-v2/issues/1139
